module-system/module-implementations/sov-attester-incentives/src/helpers.rs

```diff
@@ -0,0 +1,183 @@
+use jmt::proof::SparseMerkleProof;
+use sov_bank::{BankConfig, TokenConfig};
+use sov_modules_api::default_context::DefaultContext;
+use sov_modules_api::hooks::SlotHooks;
+use sov_modules_api::utils::generate_address;
+use sov_modules_api::{Address, Genesis, Spec};
+use sov_rollup_interface::mocks::{
+    MockBlock, MockBlockHeader, MockCodeCommitment, MockHash, MockValidityCond,
+    MockValidityCondChecker, MockZkvm,
+};
+use sov_rollup_interface::zk::ValidityConditionChecker;
+use sov_state::storage::StorageProof;
+use sov_state::{DefaultStorageSpec, ProverStorage, Storage, WorkingSet};
+
+use crate::AttesterIncentives;
+
+type C = DefaultContext;
+
+pub const TOKEN_NAME: &str = "TEST_TOKEN";
+pub const BOND_AMOUNT: u64 = 1000;
+pub const DEFAULT_CHAIN_HEIGHT: u64 = 12;
+pub const INITIAL_BOND_AMOUNT: u64 = 5 * BOND_AMOUNT;
+pub const SALT: u64 = 5;
+pub const DEFAULT_ROLLUP_FINALITY: u64 = 3;
+pub const INIT_HEIGHT: u64 = 0;
+
+/// Consumes and commit the existing working set on the underlying storage
+/// `storage` must be the underlying storage defined on the working set for this method to work.
+pub(crate) fn commit_get_new_working_set(
+    storage: &ProverStorage<DefaultStorageSpec>,
+    working_set: WorkingSet<<C as Spec>::Storage>,
+) -> WorkingSet<<C as Spec>::Storage> {
+    let (reads_writes, witness) = working_set.checkpoint().freeze();
```

Comment:
Why do we need to commit inside a module? `AppTemplate` should take care of it. Now I think we should not expose the `checkpoint` method in `working_set`.

Reply:
Ahh if this is for tests then't let's make `/test` folder and move it there.

Reply:
Done